### ISSUE TITLE
azurerm_sql_database fails to create Database

### ISSUE THREAD

**Issue by listentorick** [hashicorp/terraform-provider-azurerm#15328]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

* `azurerm_sql_database`

### Terraform Configuration Files

resource "azurerm_sql_server" "tenant_sql" {
  name                         = "fakename"
  resource_group_name          = azurerm_resource_group.tenant_rg.name
  location                     = var.location
  version                      = "12.0"
  administrator_login          = var.sql_server_admin_login
  administrator_login_password = random_password.sql_server_admin_password.result

   tags = {
        Application = "blah"
      Tenant = var.tenant
      CustomerId = var.customer_id
      Environment = var.environment
    }
}
The db is created like this

resource "azurerm_sql_database" "tenant_sqldb" {
  name                             = "x"
  resource_group_name              = azurerm_resource_group.tenant_rg.name
  location                         =  var.location
  server_name                      = "fakename"
  create_mode                      = "Default"
  edition                          = var.database_is_premium == true ? "Premium" : "Standard"
  requested_service_objective_name = var.database_is_premium == true ? "P4" : "S1"
  max_size_bytes = 268435456000
   tags = {
      Application = "bah"
      Tenant      = var.tenant
      CustomerId  = var.customer_id
      Environment = var.environment
    }
}

### Expected Behaviour

A SQL Server DB and Database are created.

### Actual Behaviour

I'm using terraform 2.33.0 (I tested 2.90.0 too ) to create a sql server and then create a database

Over the past week I've started  intermittently getting the following error:

Error waiting on create/update future for SQL Database "x" (Resource Group "y", Server "z"): Code="45157" Message="Server 'z' is busy with another operation. Please try your operation later.

I can create the SQL database in the portal fine after this failure.

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by listentorick** [comment]
FWIW I fixed this by upgrading to azurerm_mssql_server etc

**Comment by mybayern1974** [comment]
@listentorick , thanks for opening this issue. With assuming `azurerm_mssql_` is more recommended than `azurerm_sql_` and seems you can be unblocked by this, would you be okey if we close this issue?

**Comment by rcskosir** [comment]
Thanks for opening this issue. This was a problem in the 2.x version of the provider which is no longer actively maintained. 
If you have future bugs or feature requests please do let us know by opening a new issue, thanks!
